NSD Mitaka - Log meta data

pkharat opened the pull request.

pkharat committed:
Initial code check-in to have meta-data in logs in orchestrator

pkharat committed:
Merge branch 'mitaka_21st_march_base' of https://github.com/oneconvergence/group-based-policy into mitaka-logging

Conflicts:
	gbpservice/nfp/orchestrator/modules/service_orchestrator.py

pkharat committed:
1) Added meta data in the logs in orchestrator, configurator
2) Fixed orchestrator unit test cases which were failing bec of log meta data changes
3) Added new lib file nfp_log_helper.py which is helper to prepare log meta data

Commit:
Fixed firewall in N-S scenario, Added nfp in place of nsf

Commit:
Fixed delete path issue in device orchestrator

Commit:
Modified delete path error logs level as warning

pkharat committed:
1) Meta data in logs for provider/consumer pt creation/deletion

pkharat committed:
Merge branch 'mitaka_21st_march_base' of https://github.com/oneconvergence/group-based-policy into mitaka-logging

Conflicts:
	gbpservice/neutron/tests/unit/nfp/orchestrator/modules/test_service_orchestrator.py
	gbpservice/nfp/orchestrator/modules/service_orchestrator.py

pkharat committed:
Fixed merge conflict

pkharat committed:
Corrected indent argument in make_dict_readable() api

rajendramachani committed:
Fix for the stale OVS ports

The stale OVS ports on compute are because the neutron port is deleted before the nova instance is completely deleted.
Fixing this by deleting the device instance related resources only after device instance deletion is completed.

rajendramachani committed:
Fixed poll event handling

njagadish committed:
Merge pull request #213 from oneconvergence/mitaka_nfp_14_april_2016

Mitaka nfp 14 april 2016

ashutosh-mishra commented on gbpservice/nfp/lib/nfp_log_helper.py:
```diff
@@ -0,0 +1,144 @@
+                    if nf.get('service_chain_id')
+                    else ''),
+                '05_service_chain_instance_id': (
+                    ('ServiceInstanceID:' +
```
ServiceInstanceID is not clear, can we make it as NetworkFunctionID

ashutosh-mishra commented on gbpservice/nfp/lib/nfp_log_helper.py:
```diff
@@ -0,0 +1,144 @@
+                log_msg += log_info_content.get(content) + ', '
+        log_msg = log_msg[:-2]
+        if log_msg:
+            log_msg += '] - '
```
Above check is not working, attached logs are
ServiceInstanceID:04966fa1-cef1-4515-8a68-71184905619e]Sending delete NFD config request to configurator

ashutosh-mishra commented on gbpservice/nfp/orchestrator/modules/device_orchestrator.py:
```diff
@@ -83,12 +84,16 @@ def __init__(self, conf, controller):
-                     "%s(event_name)s with event data: %(event_data)s"),
-                 {'event_name': event_id, 'event_data': event_data})
+        log_meta_data = (event_data.get('log_meta_data')
+                         if 'log_meta_data' in event_data else '')
```
As we are doing get on event_data dict, no need to check for key, we can also do
event_data.get('log_meta_data', '')
Same applicable on all further checks.

ashutosh-mishra commented on gbpservice/nfp/lib/nfp_log_helper.py:
```diff
@@ -0,0 +1,144 @@
+                           it along with other params like level,event etc
+    :param level - log level
+    :param event_category
+    :param event
```
For params level, event_category, event, Can it be any string ?
For level we can define constants.

ashutosh-mishra commented on gbpservice/nfp/lib/nfp_log_helper.py:
```diff
@@ -0,0 +1,144 @@
+        log_msg = '['
+        for content in log_info_content:
+            if log_info_content.get(content):
+                log_msg += log_info_content.get(content) + ', '
```
No need to again do get, inside if check

ashutosh-mishra commented on gbpservice/nfp/lib/nfp_log_helper.py:
```diff
@@ -0,0 +1,144 @@
+    except Exception as err:
+        LOG.error(_LE("Error while generating LOG. %s."
+                      % str(err).capitalize()))
+        raise err
```
This will go as unhandled exception, instead of raising exception  we can return empty string.

ashutosh-mishra commented:
Looks like attached log file is old, Can you give latest log file
We need to remove this log file, once review is done.

rajendramachani committed:
Fixed DEVICE_SPAWNING poll event

pkharat committed:
Merge branch 'mitaka_21st_march_base' of https://github.com/oneconvergence/group-based-policy into mitaka-logging

Conflicts:
	gbpservice/nfp/orchestrator/drivers/orchestration_driver_base.py

Commit:
Adding changes made while submitting patches

hiteshhapani committed:
service vm changes

YogeshRajmane committed:
Added orchestrator side changes for base mode with vm

njagadish committed:
Merge pull request #216 from oneconvergence/base_mode_with_vm_orchestrator

Added orchestrator side changes for base mode with vm

njagadish committed:
Modified error log to debug log.

njagadish committed:
Merge pull request #215 from oneconvergence/service_vm_changes

service vm changes

njagadish committed:
Merge pull request #214 from oneconvergence/SC2-857

Fixed firewall in N-S scenario, Added nfp in place of nsf

rajendramachani committed:
fixed pep8 warnings

rajendramachani committed:
Pulled manually PR #217

njagadish committed:
Merge pull request #218 from oneconvergence/mitaka_nfp_15_april_2016

Mitaka nfp 15 april 2016

pkharat committed:
1) Used oslo messaging RPCClient to publish notifications instead of using pika
2) Corrected error in pecan's pull_notifications()

Commit:
Fixed bug found in CI run

njagadish committed:
Merge pull request #220 from oneconvergence/bug_fixes

Fixed bug found in CI run

njagadish committed:
Merge pull request #222 from oneconvergence/mitaka_nfp_16_april_2016

Mitaka nfp 16 april 2016

pkharat committed:
Merge branch 'mitaka_21st_march_base' of https://github.com/oneconvergence/group-based-policy into mitaka-logging

Conflicts:
	gbpservice/nfp/orchestrator/modules/device_orchestrator.py

rajendramachani committed:
Orchestration driver API changes

pkharat committed:
Reorganized the code

pkharat committed:
Reverting the push and pull notifications to old method of
using python queue instead of using rabbitmq as pika hanging pecan server

njagadish committed:
Merge pull request #227 from oneconvergence/nsd-controller-socket-error-fix

Nsd controller socket error fix

njagadish committed:
Merge pull request #228 from oneconvergence/mitaka_nfp_19_april_2016

Mitaka nfp 19 april 2016

Commit:
added base mode and base mdoe with VM

AkashDeepSrivastava committed:
committed api changes for the base mode

YogeshRajmane committed:
Added API changes from orchestrator side

YogeshRajmane committed:
Fixed PT add/remove and consumer add/remove path

rajendramachani committed:
Orchestrator driver API changes.

rajendramachani committed:
Modified info in the config_orch api

YogeshRajmane committed:
Fixed test case errors

mak-454 committed:
Oslo local context expects monkey_patching to be done before importing it, corrected it

rajendramachani committed:
Merge pull request #231 from oneconvergence/orchestration_driver_api_changes_merge

Orchestration driver api changes

Commit:
review comments addressed

Commit:
Merge branch 'basemode-api-changes-integration' of https://github.com/oneconvergence/group-based-policy into basemode-api-changes-integration

njagadish committed:
Merge pull request #232 from oneconvergence/base_mode_stress_test_fix

Fixed oslo local context exception

Commit:
Added changes a/c to review comment

Commit:
Added new service tag custom_json for base mode with service vm

rajendramachani committed:
Fixed orchestrator unit tests

Commit:
Added changes a/c to review comment by ahmed

Commit:
Merge branch 'basemode-api-changes-integration' of https://github.com/oneconvergence/group-based-policy into basemode-api-changes-integration

YogeshRajmane committed:
fixed pep8 errors

YogeshRajmane committed:
Merge branch 'basemode-api-changes-integration' of https://github.com/oneconvergence/group-based-policy into basemode-api-changes-integration

YogeshRajmane committed:
Fixed integration issue

Commit:
Fixed integration issue

Commit:
Merge branch 'basemode-api-changes-integration' of https://github.com/oneconvergence/group-based-policy into basemode-api-changes-integration

Commit:
Fixed integration issues

pkharat committed:
1) nfp_log_helper.py - Corrected log format when log_meta_data string is passed
   to prepare_log_meta_data()

Commit:
fixed notification path

YogeshRajmane committed:
Fixed pep8 errors from dummy service vm code

dpaks committed:
increased timeout for proxy and base mode pecan

dpaks committed:
Merge branch 'basemode-api-changes-integration' of https://github.com/oneconvergence/group-based-policy into basemode-api-changes-integration

dpaks committed:
pep8 fixes

dpaks committed:
added PTG creation/deletion scripts for base mode for LB and base mode with VM for FW

dpaks committed:
renamed base mode PTG creation scripts

njagadish committed:
Moved the base mode VM code to conftrib directory.

dpaks committed:
added cleanup script for fw basemode with vm

dpaks committed:
Merge branch 'basemode-api-changes-integration' of https://github.com/oneconvergence/group-based-policy into basemode-api-changes-integration

njagadish committed:
Added latest script that does the iptable rule configuration.

njagadish committed:
Merge branch 'basemode-api-changes-integration' of https://github.com/oneconvergence/group-based-policy into basemode-api-changes-integration

YogeshRajmane committed:
Fixed pep8 errors of configure_fw_rules

njagadish committed:
Merge pull request #229 from oneconvergence/basemode-api-changes-integration

added base mode and base mode with VM

njagadish committed:
Merge pull request #233 from oneconvergence/mitaka_nfp_20_april_2016

Mitaka nfp 20 april 2016

Commit:
Changes w.r.t reference_configurator path

Commit:
Corrected path

njagadish committed:
Merge pull request #234 from oneconvergence/basemode-api-changes-integration

Changes w.r.t reference_configurator path

njagadish committed:
Merge pull request #235 from oneconvergence/mitaka_nfp_20_april_2016

Mitaka nfp 20 april 2016 - reference configurator scritps and bins

YogeshRajmane committed:
Fixed pep8 error in reference configurator

njagadish committed:
Merge pull request #236 from oneconvergence/mitaka_nfp_20_april_2016

Fixed pep8 error in reference configurator

rajendramachani committed:
devstack and diskimage-create scripts

rajendramachani committed:
added DISABLE_IMAGE_BUILDER

rajendramachani committed:
Modified SERVICE_GIT path and fixed pep8 warnings

njagadish committed:
Merge pull request #237 from oneconvergence/devstack-and-diskimage-builder

devstack and diskimage-create scripts

njagadish committed:
Merge pull request #238 from oneconvergence/mitaka_nfp_21_april_2016

Mitaka nfp 21 april 2016

rajendramachani committed:
Added missing file

rajendramachani committed:
Merge pull request #239 from oneconvergence/devstack-and-diskimage-builder

Added missing file

rajendramachani committed:
Merge pull request #240 from oneconvergence/mitaka_nfp_21_april_2016

Mitaka nfp 21 april 2016

rajendramachani committed:
Fixed pep8 errors

dpaks committed:
merged advanced mode code to the latest daily branch

AkashDeepSrivastava committed:
added changes for review comment

hiteshhapani committed:
Fixed pep8 errors

hiteshhapani committed:
Merge branch 'advancedmode-api-changes-22-april' of https://github.com/oneconvergence/group-based-policy into advancedmode-api-changes-22-april

dpaks committed:
removed tenant and service ids

AkashDeepSrivastava committed:
updated changes observed while testing firewall on 22ndapril

AkashDeepSrivastava committed:
Merge branch 'advancedmode-api-changes-22-april' of https://github.com/oneconvergence/group-based-policy into advancedmode-api-changes-22-april

rajendramachani committed:
Fixed diskimage-create scripts

rajendramachani committed:
Merge pull request #241 from oneconvergence/pep8_correction_in_diskimage_create_scripts

Fixed pep8 errors in diskimage_create scripts

rajendramachani committed:
Pulled PR #243 manually

rajendramachani committed:
Merge pull request #244 from oneconvergence/mitaka_nfp_22_april_2016

Mitaka nfp 22 april 2016

dpaks committed:
bug fixes for loadbalancer

dpaks committed:
bug fixes for demuxer

AkashDeepSrivastava committed:
fixed bugs during debugging

AkashDeepSrivastava committed:
Merge branch 'advancedmode-api-changes-22-april' of https://github.com/oneconvergence/group-based-policy into advancedmode-api-changes-22-april

dpaks committed:
removed use of omittable keys in accordance with the blue print. E.g., rule_info, active*

dpaks committed:
Merge branch 'advancedmode-api-changes-22-april' of https://github.com/oneconvergence/group-based-policy into advancedmode-api-changes-22-april

rajendramachani committed:
Few more fixes

rajendramachani committed:
Merge pull request #245 from oneconvergence/diskimage-create-integration-test-fixes

Few more fixes - diskimage-create integration

rajendramachani committed:
Merge pull request #246 from oneconvergence/mitaka_nfp_22_april_2016

Mitaka nfp 22 april 2016

YogeshRajmane committed:
Fixed default routes adding issue

njagadish committed:
Merge pull request #247 from oneconvergence/mitaka_nfp_23_april_2016

Fixed default routes adding issue

YogeshRajmane committed:
Adding ICMP accept action in script

njagadish committed:
Merge pull request #248 from oneconvergence/mitaka_nfp_23_april_2016

Adding ICMP accept action in script

pkharat committed:
Merge branch 'mitaka_21st_march_base' of https://github.com/oneconvergence/group-based-policy into mitaka-logging

Conflicts:
	gbpservice/neutron/tests/unit/nfp/orchestrator/modules/test_device_orchestrator.py
	gbpservice/nfp/orchestrator/drivers/orchestration_driver_base.py
	gbpservice/nfp/orchestrator/modules/device_orchestrator.py
	gbpservice/nfp/orchestrator/modules/service_orchestrator.py

sdharav committed:
vpn changes

sdharav committed:
changes according to the new data format

dpaks committed:
fixed unit tests that broke due to the API changes

pkharat committed:
Corrected log_meta_data changes according new API changes

Commit:
Added Node update support in Orchestrator

njagadish committed:
Merge pull request #250 from oneconvergence/advancedmode-api-changes-22-april

Advancedmode api changes 22 april

Commit:
Corrected condition for negative cases

AkashDeepSrivastava committed:
bug fixes for unix socket

AkashDeepSrivastava committed:
Merge branch 'mitaka_nfp_26_april_2016' of https://github.com/oneconvergence/group-based-policy into mitaka_nfp_26_april_2016

sdharav committed:
fixes for vpn datapath

mak-454 committed:
Merge pull request #251 from oneconvergence/mitaka_nfp_26_april_2016

Mitaka nfp 26 april 2016

Commit:
Corrected condition for negative cases, if any of them not exist don't proceed

pkharat committed:
1) Added lib function in nfp_log_helper to get log_meta_data from api
2) Accordingly in the utc modules corrected way of getting log_meta_data from api

YogeshRajmane committed:
Fixed multiple PT add in LB

sdharav committed:
Merge branch 'mitaka_21st_march_base' of https://github.com/oneconvergence/group-based-policy into advancedmode-api-changes-22-april-vpn

sdharav committed:
resttime out changes

YogeshRajmane committed:
Corrected log message

pkharat committed:
Merge branch 'mitaka_21st_march_base' of https://github.com/oneconvergence/group-based-policy into mitaka-logging

Conflicts:
	gbpservice/nfp/configurator/agents/agent_base.py
	gbpservice/nfp/configurator/agents/nfp_service.py
	gbpservice/nfp/configurator/drivers/firewall/vyos/vyos_fw_driver.py

YogeshRajmane committed:
Removed rule info while sending to configurator

ashutosh-mishra committed:
Addressed review comments

dpaks committed:
removed usage of rule info from configurator

AkashDeepSrivastava committed:
corrected config_orch unit test and fixes bug in vpn path

sdharav committed:
vpn datapath fixes and constants

AkashDeepSrivastava committed:
revert my changes in vpn file as it is already changed in 259 pull request

AkashDeepSrivastava committed:
fixed pep8 error in vpn.py

njagadish committed:
Merge pull request #259 from oneconvergence/advancedmode-api-changes-22-april-vpn

VPN - As per advanced mode api changes

mak-454 committed:
Merge pull request #257 from oneconvergence/mitaka_nfp_27_april_2016_with_config_orch_unit_test_and_vpn_path_fix

Changes look ok.

njagadish committed:
Merge pull request #256 from oneconvergence/eliminate_rule_info

Eliminated rule info

njagadish committed:
Merge pull request #255 from oneconvergence/lb_multiple_pt_fix

Lb multiple pt fix

dpaks committed:
Addressed review comments

njagadish committed:
Merge pull request #262 from oneconvergence/node_update

Node update

dpaks committed:
replaced service_type to resource_type

dpaks committed:
removed rule info

YogeshRajmane committed:
Synced with submitted patch

YogeshRajmane committed:
Added service_vendor as nfp code changes

YogeshRajmane committed:
Removed not required comment

YogeshRajmane committed:
Added missing instruction in README

dpaks committed:
updated comments

YogeshRajmane committed:
Changed patch number with latest 28

njagadish committed:
Merge pull request #265 from oneconvergence/mitaka_nfp_27_april_2016

Mitaka nfp 27 april 2016

rajendramachani committed:
Fixed DOMAIN vars requirement for flavor create

dpaks committed:
rebased to nfp 29 daily branch

rajendramachani committed:
Fixed steps in README-NFP

dpaks committed:
removed config option for rest timeout

njagadish committed:
Merge pull request #268 from oneconvergence/service_vendor_nfp_for_demo

Devstack changes for demo

dpaks committed:
removed unused cfg import from vpn agent

njagadish committed:
Merge pull request #267 from oneconvergence/configurator-review-comments-addressal-nfp29

Addressed configurator review comments

sdharav committed:
vpn service driver

sdharav committed:
removed dev debug

sdharav committed:
removed unwanted code

dpaks committed:
synced with nfp29 daily branch

dpaks committed:
added config opts file in lib directory

dpaks committed:
renamed config option parameters

dpaks committed:
visibility API in svc vm agents

njagadish committed:
Merge pull request #271 from oneconvergence/visibilty-api-in-genericconfig-driver-nfp29-new

Visibilty APIs in generic config driver

dpaks committed:
removed the term visibility

sdharav committed:
addressed the review comments

sdharav committed:
renamed the vpnaas driver

sdharav committed:
changed paramters

njagadish committed:
Merge pull request #272 from oneconvergence/visibilty-api-in-svc-vm-agents

Visibilty API in Svc VM agents

Commit:
Corrected logging option to include http logs also

sdharav committed:
addressed review comments

sdharav committed:
changed the vpn topc

sdharav committed:
vpn review comments addressed

rajendramachani committed:
Merge pull request #270 from oneconvergence/mitaka_nfp_29_april_2016_vpn_servicedriver

Mitaka nfp 29 april 2016 vpn service driver

njagadish committed:
Merge pull request #273 from oneconvergence/sc2-653

Fixed logging option in haproxy, Corrected logging option to include http logs also

Commit:
Added missed wait time for pt, ptg operations, added fix for stack not cleaned up issue.

dpaks committed:
made HM and log 4wd common

dpaks committed:
comments fix

dpaks committed:
comments fix

dpaks committed:
removed unnecessary constant

njagadish committed:
Merge pull request #275 from oneconvergence/configurator_cleanup

Configurator enhancements

Commit:
Fixed pep8 errors

njagadish committed:
Merge pull request #279 from oneconvergence/mitaka_nfp_29_april_2016

Mitaka nfp 29 april 2016

Commit:
Merge branch 'mitaka_21st_march_base' of https://github.com/oneconvergence/group-based-policy into sc2-877

AkashDeepSrivastava committed:
fix for none value of service_vendor for network_function_devices db

njagadish committed:
Merge pull request #281 from oneconvergence/mitaka_nfp_05_may_2016_service_vendor_fix

fix for none value of service_vendor for network_function_devices db

njagadish committed:
Merge pull request #280 from oneconvergence/sc2-877

Added missed wait time for pt, ptg operations, added fix for stack not cleaned up issue.

njagadish committed:
Merge pull request #282 from oneconvergence/mitaka_nfp_05_may_2016

Mitaka nfp 05 may 2016

pkharat committed:
1) Log meta data changes in configurator
2) Restructured functions in nfp_log_helper.py
3) Changed orchestrator code according to new design of log meta data

pkharat committed:
Merge branch 'mitaka_21st_march_base' of https://github.com/oneconvergence/group-based-policy into mitaka-logging

Conflicts:
	gbpservice/nfp/configurator/drivers/base/base_driver.py
	gbpservice/nfp/configurator/drivers/firewall/vyos/vyos_fw_driver.py
	gbpservice/nfp/configurator/drivers/loadbalancer/v1/haproxy/haproxy_lb_driver.py
	gbpservice/nfp/lib/transport.py
	gbpservice/nfp/orchestrator/modules/device_orchestrator.py
	gbpservice/nfp/orchestrator/modules/service_orchestrator.py

pkharat committed:
Updated log changes for member add/delete

njagadish commented:
Approach of logging changed and those changes are merged to mainline. This PR is not needed. Closing it.
